components/org.wso2.carbon.identity.application.authenticator.push/src/main/java/org/wso2/carbon/identity/application/authenticator/push/PushAuthenticator.java

```diff
@@ -186,19 +197,32 @@ protected void processAuthenticationResponse(HttpServletRequest httpServletReque
             throw new AuthenticationFailedException(errorMessage, e);
         }
         if (claimsSet != null) {
-            validateChallenge(claimsSet, serverChallenge, deviceId);
-
-            String authStatus =
-                    getClaimFromClaimSet(claimsSet, PushAuthenticatorConstants.TOKEN_RESPONSE, deviceId);
+            if (validator.validateChallenge(claimsSet, serverChallenge, deviceId)) {
+                String authStatus;
+                try {
+                    authStatus =
+                            validator.getClaimFromClaimSet(claimsSet, PushAuthenticatorConstants.TOKEN_RESPONSE,
+                                    deviceId);
+                } catch (PushAuthTokenValidationException e) {
+                    String errorMessage = String.format("Error in getting claim %s from the auth response token" +
+                            " received from device: %s", PushAuthenticatorConstants.TOKEN_SESSION_DATA_KEY, deviceId);
```

Comment:
The claim used in `getClaimFromClaimSet` and the claim in the error message are different.

Reply:
Fixed with https://github.com/wso2-extensions/identity-outbound-auth-push/pull/118/commits/061f61e9ccdcddd46a1a152bc5f126ed13cbd1aa